x-pack/plugins/alerting/server/lib/process_alerts.ts

```diff
@@ -16,8 +16,10 @@ interface ProcessAlertsOpts<
 > {
   alerts: Record<string, Alert<State, Context>>;
   existingAlerts: Record<string, Alert<State, Context>>;
+  previouslyRecoveredAlerts: Record<string, Alert<State, Context>>;
```

Comment:
I think the idea with this parameter is to pass in the alerts which are not active now, and recovered previously, but we still need to track the flapping history in case they _become_ flapping in the future.  I was wondering how we'd structure this, and this seems about right.